pkg/services/libraryelements/api.go

```diff
@@ -122,6 +168,14 @@ func (l *LibraryElementService) getHandler(c *contextmodel.ReqContext) response.
 		return toLibraryElementError(err, "Failed to get library element")
 	}
 
+	allowed, err := l.evalPermission(c, "read", element.UID, element.FolderID)
+	if err != nil {
+		return toLibraryElementError(err, "Failed to evaluate permissions")
+	}
+	if !allowed {
+		return response.Error(http.StatusForbidden, "You do not have the permissions needed to view this library element", nil)
+	}
+
```

Comment:
Same this is already handled at middleware layer by `authorize(ac.EvalPermission(ActionLibraryPanelsRead, uidScope)`
```suggestion
```